scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -279,3 +281,70 @@ def get_generated_keys(self) -> List[str]:
         for operand in self._operands:
             generated_keys += operand.get_generated_keys()
         return generated_keys
+
+
+class AttackPointIteratorCartesianProduct(AttackPointIterator):
+    """
+    Attack point iterator cartesian product class. This class takes any amount
+    of operands and combines them just like a cartesian product would.
+    """
+
+    def __init__(self, operation: str, operands: List[Dict[str, Any]]) -> None:
+        """Initialize the cartesian product iterator.
+          
+          Args:
+            operation (str): The operation of the iterator
+                represents what the iterator does and what 
+                has to be in the config file. This is only used once to
+                double check if the operation is the correct one.
+                
+            operands (List[Dict[str, Any]]): The operands are any number of
```

Comment:
- Zero is a number, but it still raises. This is not documented.
- What type of exception/error is raised?